controllers/render_template.go

```diff
@@ -43,31 +47,47 @@ func (r *CommonServiceReconciler) getNewConfigs(cs *unstructured.Unstructured) (
 	var sizeConfigs []interface{}
 	switch cs.Object["spec"].(map[string]interface{})["size"] {
 	case "small":
-		sizeConfigs, err = applySizeTemplate(cs, size.Small)
+		sizeConfigs, err = applySizeTemplate(cs, size.Small, isScoped)
 		if err != nil {
 			return sizeConfigs, err
 		}
 	case "medium":
-		sizeConfigs, err = applySizeTemplate(cs, size.Medium)
+		sizeConfigs, err = applySizeTemplate(cs, size.Medium, isScoped)
 		if err != nil {
 			return sizeConfigs, err
 		}
 	case "large":
-		sizeConfigs, err = applySizeTemplate(cs, size.Large)
+		sizeConfigs, err = applySizeTemplate(cs, size.Large, isScoped)
 		if err != nil {
 			return sizeConfigs, err
 		}
 	default:
-		if cs.Object["spec"].(map[string]interface{})["services"] != nil {
-			sizeConfigs = cs.Object["spec"].(map[string]interface{})["services"].([]interface{})
-		}
+		sizeConfigs = applySizeConfigs(cs, isScoped)
 	}
 	newConfigs = append(newConfigs, sizeConfigs...)
 
 	return newConfigs, nil
 }
 
-func applySizeTemplate(cs *unstructured.Unstructured, sizeTemplate string) ([]interface{}, error) {
+func applySizeConfigs(cs *unstructured.Unstructured, isScoped bool) []interface{} {
+	var dest []interface{}
+	if cs.Object["spec"].(map[string]interface{})["services"] != nil {
+		for _, configSize := range cs.Object["spec"].(map[string]interface{})["services"].([]interface{}) {
+			if !isScoped {
+				for _, operator := range clusterScopeOperators {
+					if configSize.(map[string]interface{})["name"].(string) != operator {
+						continue
```

Comment:
This function applies when there is no overall size configured, all the size settings depend on the individual services configuration.

For single instance case, all the CommonService CRs are considered as in scope. Therefore, it will not cause incompatibility to the previous release.
For the multi instances case, we want to get all the configurations from the CommonService CR which is inside the scope. But we only want the configurations about cluster scope operators from the CR which is out of the scope.

Since each cs-operator will reconcile every CommonService CR, we need to exclude the namespaced scope configs from the CR which is out of scope.

When some resources are not `ibm-cert-manager` or `licensing`, they should not append into the new ConfigSize, we use `continue` to skip it

Reply:
yes, we need to add a label to skip the outer loop.

```
OUTER:
for _, item1 := range list1 {
	for _, item2 := range list2 {
		if ... {
			continue OUTER
		}
		... 
	}
	... 
}
```